src/backend/distributed/test/progress_utils.c

```diff
@@ -0,0 +1,147 @@
+/*-------------------------------------------------------------------------
+ *
+ * progress_utils.c
+ *
+ * This file contains functions to exercise progress monitoring functionality
+ * within Citus.
+ *
+ * Copyright (c) 2017, Citus Data, Inc.
+ *
+ *-------------------------------------------------------------------------
+ */
+
+
+#include "postgres.h"
+#include "miscadmin.h"
+#include "fmgr.h"
+#include "funcapi.h"
+
+#include <unistd.h>
+
+#include "distributed/multi_progress.h"
+#include "nodes/execnodes.h"
+#include "utils/tuplestore.h"
+
+
+PG_FUNCTION_INFO_V1(create_progress);
+PG_FUNCTION_INFO_V1(update_progress);
+PG_FUNCTION_INFO_V1(finish_progress);
+PG_FUNCTION_INFO_V1(show_progress);
+
+
+Datum
+create_progress(PG_FUNCTION_ARGS)
+{
+	uint64 magicNumber = PG_GETARG_INT64(0);
+	int stepCount = PG_GETARG_INT32(1);
+	ProgressMonitorData *monitor = CreateProgressMonitor(magicNumber, stepCount,
+														 sizeof(uint64), 0);
+	uint64 *steps = (uint64 *) monitor->steps;
+
+	int i = 0;
+	for (; i < stepCount; i++)
+	{
+		steps[i] = 0;
+	}
+
+	PG_RETURN_VOID();
+}
+
+
+Datum
+update_progress(PG_FUNCTION_ARGS)
+{
+	uint64 step = PG_GETARG_INT64(0);
+	uint64 newValue = PG_GETARG_INT64(1);
+
+	ProgressMonitorData *monitor = GetCurrentProgressMonitor();
+	uint64 *steps = (uint64 *) monitor->steps;
```

Comment:
It's probably not necessary to check for NULL here since it's a test, but might still be useful in case someone copies this pattern.